src/Raven.Client/Documents/Session/AsyncSessionDocumentTimeSeries.cs

```diff
@@ -96,9 +101,11 @@ await ServeFromCacheOrGetMissingPartsFromServerAndMerge(from ?? DateTime.MinValu
                     new GetTimeSeriesOperation<TTValues>(DocId, Name, from, to, start, pageSize), Session._sessionInfo, token: token)
                 .ConfigureAwait(false);
 
-            if (rangeResult == null)
+            if ( (rangeResult == null) || ((rangeResult.Entries.Length == 0) && ( start > 0)))
```

Comment:
If I requested a range that has no entries, we will not put that into the cache, right? why?

Reply:
only when start is more then 0 because we can't know if we have entries when start will be 0

Reply:
but we can know that on the server when returning the results, and change the `From` accordingly.

Reply:
Also to consider is that if I'm doing the same request twice, I should go only once to the server.
But if I have `start` in my request, will that make me for the same request go twice to the server?